src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```diff
@@ -149,6 +149,26 @@ def validate_filter_parameter(string):
     return result
 
 
+def validate_assigned_identity(namespace):
+    if namespace.assign_identity:
+        from msrestazure.tools import is_valid_resource_id
+
+        namespace.assign_identity = set(namespace.assign_identity)
+        for identity in namespace.assign_identity:
+            if identity != '[system]' and not is_valid_resource_id(identity):
+                raise CLIError("Invalid identity '{}'. Use '[system]' to refer system assigned identity, or a resource id to refer user assigned identity.".format(identity))
+
+
+def validate_identities(namespace):
+    if namespace.identities:
+        from msrestazure.tools import is_valid_resource_id
+
+        namespace.identities = set(namespace.identities)
+        for identity in namespace.identities:
+            if identity != '[system]' and identity != '[all]' and not is_valid_resource_id(identity):
```

Comment:
> and identity != '[all]' [](start = 37, length = 24)

This case should only be valid for the remove identity command right?

Reply:
Yes. there is a validation in assign_identity to exlcude '[all]'. We can have two validators for assign and remove identity, but they will be very similar except for '[all]' check. 

---
In reply to: [364514597](https://github.com/Azure/azure-cli/pull/11795#discussion_r364514597) [](ancestors = 364514597)